test/gpu/octree/test_knn_search.cpp

```diff
@@ -149,8 +154,12 @@ TEST(PCL_OctreeGPU, exactNeighbourSearch)
         int beg = i * k;
         int end = beg + k;
 
+        const std::vector<float>::iterator beg_dist2 = downloaded_sqr_dists.begin() + beg;
+        const std::vector<float>::iterator end_dist2 = downloaded_sqr_dists.begin() + end;
+
         downloaded_cur.assign(downloaded.begin() + beg, downloaded.begin() + end);
-        
+        downloaded_sqr_dists_cur.assign(beg_dist2, end_dist2);
```

Comment:
There's a constructor which allows you to specify a range to initialize from.
https://en.cppreference.com/w/cpp/container/vector/vector number 5 on that list.